src/gadgets/num.rs

```diff
@@ -407,6 +408,43 @@ impl<E: ScalarEngine> Num<E> {
             lc: self.lc + &bit.lc(one, coeff),
         }
     }
+
+    pub fn add(self, other: &Self) -> Self {
```

Comment:
can you add some basic tests that these two methods do what you expect them to do please?

Reply:
I actually put them next to the existing tests.

All right, all right.

Reply:
Done.